cpp/src/arrow/filesystem/path_util.h

```diff
@@ -75,6 +75,9 @@ util::string_view RemoveTrailingSlash(util::string_view s);
 ARROW_EXPORT
 Status AssertNoTrailingSlash(util::string_view s);
 
+ARROW_EXPORT
+Status AssertLeadingSlash(util::string_view s);
```

Comment:
Should this be just return bool? It is a bit doubtful and tried to maintain the convention already in use.

cc @westonpace

Reply:
Yes, it should probably return a bool and be named `HasLeadingSlash` but, at that point, the method is just...

```
bool HasLeadingSlash(util::string_view key) {
  return key.front() == '/';
}
```

...and I'm not really sure such a method would be really needed.